pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/TaskGeneratorUtils.java

```diff
@@ -68,4 +73,30 @@
     }
     return runningSegments;
   }
+
+  public static Map<String, List<String>> getScheduledSegmentsMap(@Nonnull String taskType,
```

Comment:
(nit) Don't use the non-null annotation (you can also remove them from other places), we only use nullable for readability.